addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/PiiScanRuleUnitTest.java

```diff
@@ -400,6 +422,100 @@ void shouldRaiseAlertWhenPiiInJavascriptContentTypeOnLOWThreshold() throws Excep
         assertThat(alertsRaised.size(), is(1));
     }
 
+    private static Stream<Arguments> provideContentAndThresholds() {
+        return Stream.of(
+                Arguments.of(CONTENT_INSIDE_SCRIPT_TAG, AlertThreshold.MEDIUM),
+                Arguments.of(CONTENT_INSIDE_IMG_ID_ATTR, AlertThreshold.MEDIUM),
+                Arguments.of(CONTENT_INSIDE_H1_CLASS_ATTR, AlertThreshold.MEDIUM),
+                Arguments.of(CONTENT_INSIDE_HTML_COMMENT, AlertThreshold.MEDIUM),
+                Arguments.of(CONTENT_INSIDE_SCRIPT_TAG, AlertThreshold.HIGH),
+                Arguments.of(CONTENT_INSIDE_IMG_ID_ATTR, AlertThreshold.HIGH),
+                Arguments.of(CONTENT_INSIDE_H1_CLASS_ATTR, AlertThreshold.HIGH),
+                Arguments.of(CONTENT_INSIDE_HTML_COMMENT, AlertThreshold.HIGH));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideContentAndThresholds")
+    void shouldNotRaiseAlertWhenPiiIsInNonVisibleHtmlContentTypeOnNonLowThreshold(
+            String content, AlertThreshold threshold) throws Exception {
+        // Given
+        HttpMessage msg = createMsg("4743251660071187");
+        msg.setResponseBody(content);
+        msg.getRequestHeader().setURI(new URI("http://example.com/generate/", true));
+        rule.setAlertThreshold(threshold);
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised, is(empty()));
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                CONTENT_INSIDE_SCRIPT_TAG,
+                CONTENT_INSIDE_IMG_ID_ATTR,
+                CONTENT_INSIDE_H1_CLASS_ATTR,
+                CONTENT_INSIDE_HTML_COMMENT
+            })
+    void shouldRaiseAlertWhenPiiInNonVisibleHtmlContentTypeOnLowThreshold(String content)
+            throws Exception {
+        String ccNum = "4743251660071187";
+        HttpMessage msg = createMsg(ccNum);
+        msg.setResponseBody(content);
+        msg.getRequestHeader().setURI(new URI("http://example.com/generate/", true));
+        rule.setAlertThreshold(AlertThreshold.LOW);
+        // When
+        scanHttpResponseReceive(msg);
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+        assertThat(alertsRaised.get(0).getEvidence(), is(equalTo(ccNum)));
+    }
+
+    @ParameterizedTest
+    @EnumSource(names = {"HIGH", "MEDIUM", "LOW"})
+    void shouldAlertOnPiiInScriptBlockRegardlessOfThreshold(AlertThreshold threshold)
```

Comment:
- These and following should follow the same naming scheme for consistency.
- The name should have `WithoutUnderscore`.